resources/definitions/smoothie_dual.def.json

```diff
@@ -0,0 +1,54 @@
+{
+    "version": 2,
+    "name": "Smoothie Dual Custom",
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": true,
+        "author": "grk3010",
+        "manufacturer": "Custom",
+        "file_formats": "text/x-gcode",
+        "has_machine_quality": true,
+        "has_variants": true,
+        "preferred_variant_name": "Mosquito Magnum 0.8",
+        "preferred_quality_type": "normal",
+        "variants_name": "Nozzle",
+        "machine_extruder_trains":
+        {
+            "0": "bondtech_bmgx2m_extruder_0",
+            "1": "bondtech_bmgx2m_extruder_1"
+        }
+    },
+    "overrides": {
+        "machine_name": { "default_value": "Smoothie Dual Custom" },
+        "machine_width": {
+            "default_value": 600
+        },
+        "machine_depth": {
+            "default_value": 550
+        },
+        "machine_height": {
+            "default_value": 500
+        },
+        "machine_center_is_zero": {
+            "default_value": false
+        },
+        "machine_gcode_flavor": {
+            "default_value": "RepRap (Marlin/Sprinter)"
+        },
+        "machine_always_write_active_tool": {
+        	"default_value": true
+        },
+        "machine_extruder_count": {
+            "default_value": 2
+        },
+        "machine_heat_zone_length": {
+        	"default_value": 15
+        },
+        "machine_start_gcode": {
+            "default_value": "G28 X0 Y0 ; Home X Y\nM375; Load Bed Leveling Grid\nG1 X300 Y275 F15000 ; Move to bed center\nM280 S3.0 ; Deploy probe pin\n## NOTE: Adjust \"ZX.XX\" after G30 to be the Z-offset (in mm) between the probe and the nozzle\n## This is how you adjust nozzle printing height!\nG30 Z1.25 ; Move down to find bed & set Z home offset\nM280 S7.0 ; Retract probe\nG1 Z15.0 F600 ; move extruder up 15mm\nT0; Activate Extruder 1"
+        },
+        "machine_end_gcode": {
+            "default_value": "G28 X0 Y0; Home X and Y\nM104 S0 ; turn off extruder\nM140 S0 ; turn off bed\nM107; turn off fans\nM84 ; disable motors"
+        }
```

Comment:
As far as I understand, this printer is intended to be a "custom" type printer, which could be any printer but with the motion controller swapped out for a Smoothieboard. It's better then to not override any of these settings, but keep them at the fdmprinter defaults.

The Smoothieboard doesn't magically make your printer 600x550x500mm, or give it 2 extruders. I'd expect this printer to only override the g-code flavour, machine_always_write_active_too, and perhaps the start and end g-code. Take a look at how [Custom FFF Printer](https://github.com/Ultimaker/Cura/blob/master/resources/definitions/custom.def.json) looks, for example.

Reply:
That make sense. I made all the configuration files match my current printer configuration for testing purposes. I'll make the changes to make it more generic.